evm_loader/lib/src/account_storage.rs

```diff
@@ -289,13 +289,40 @@ impl<'rpc, T: Rpc + BuildConfigSimulator> EmulatorAccountStorage<'rpc, T> {
 }
 
 impl<'a, T: Rpc> EmulatorAccountStorage<'_, T> {
+    async fn apply_state_overrides(&self, state_overrides: &AccountOverrides) -> NeonResult<()> {
+        for (address, overrides) in state_overrides.into_iter() {
+            if let Some(nonce) = overrides.nonce {
+                let target_chain_id = self
+                    .contract_chain_id(*address)
```

Comment:
It is the primary chain_id for contracts. For this logic it is unuseful. You need to get the current chain_id value from emulate parameters https://github.com/neonlabsorg/neon-evm/blob/768c0bdc767295b1b13f7ec7de0f0210af0498d2/evm_loader/lib/src/types/mod.rs#L53

Reply:
done